pkg/controllers/osc/resources/operating_system_config.go

```diff
@@ -144,11 +161,70 @@ type filesData struct {
 	CloudConfig           string
 	ContainerRuntime      string
 	ContainerdVersion     string
-	CloudProviderName     types.CloudProvider
+	CloudProviderName     providerconfigtypes.CloudProvider
+	NetworkConfig         providerconfigtypes.NetworkConfig
 	ExtraKubeletFlags     []string
 	ExternalCloudProvider bool
 	PauseImage            string
 	InitialTaints         string
+	OperatingSystemConfig
+}
+
+type OperatingSystemConfig struct {
+	AmazonLinuxConfig amzn2.Config
+	CentOSConfig      centos.Config
+	FlatcarConfig     flatcar.Config
+	RhelConfig        rhel.Config
+	SlesConfig        sles.Config
+	UbuntuConfig      ubuntu.Config
+}
+
+func setOperatingSystemConfig(os providerconfigtypes.OperatingSystem, operatingSystemSpec runtime.RawExtension, data *filesData) error {
```

Comment:
So probably this won't work in the future since we are getting rid of the user data plugin completely in the machine controller. Thus, I can suggest copying over each user data plugin that we have in machine controller to the `providerconfig` package(./pkg/providerconfig/[os-name]`

Reply:
Done